app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.statistics
+
+import android.support.annotation.WorkerThread
+import com.duckduckgo.app.statistics.VariantManager.VariantFeature.DefaultBrowserFeature.ShowInOnboarding
+import com.duckduckgo.app.statistics.VariantManager.VariantFeature.DefaultBrowserFeature.ShowTimedReminder
+import com.duckduckgo.app.statistics.store.StatisticsDataStore
+import timber.log.Timber
+
+@WorkerThread
+interface VariantManager {
+
+    sealed class VariantFeature {
+
+        sealed class DefaultBrowserFeature : VariantFeature() {
+            object ShowInOnboarding : DefaultBrowserFeature()
+            object ShowTimedReminder : DefaultBrowserFeature()
+        }
+    }
+
+    companion object {
+
+        // there must always be at least one active variant defined here
```

Comment:
Mimicking existing behaviour I guess - where there was always a variant code in use `ma`. I think there might be a convention server-side to use underscore to denote _no active variant_.

So if the list of active variants is empty, we could fallback to `m_` as the variant code. Or would you expect us to _not_ submit any variant information in that case?

Reply:
This PR supersedes existing behavior. In the past we always had a variant but now we only want to add them for active experiments. When we are not running an experiment there should be no active variants.

Reply:
As per discussion on a call, we will ensure list can be empty.

Furthermore, we will treat the case whereby a user is in no variant as them having `""` an empty string for their variant code. This will preserve existing logic of appending it onto the ATB version but will obviously not add anything on when there is no variant in use.